docs/transactions-new.md

```diff
@@ -0,0 +1,749 @@
+# Transactions
+
+* [General Information](#general-information)
+* [Genesis Transactions](#genesis-transactions)
+* [Common Structure](#common-structure)
+* [Domain Ledger](#domain-ledger)
+
+    * [NYM](#nym)    
+    * [ATTRIB](#attrib)    
+    * [SCHEMA](#schema)
+    * [CLAIM_DEF](#claim_def)
+    
+* [Pool Ledger](#pool-ledger)    
+    * [NODE](#node)
+    
+* [Config Ledger](#config-ledger)    
+    * [POOL_UPGRADE](#pool_upgrade)
+    * [NODE_UPGRADE](#node_upgrade)
+    * [POOL_CONFIG](#pool_config)
+
+## General Information
+
+This doc is about supported transactions and their representation on the Ledger (that is internal one).
+If you are interested in the format of client's Request (both write and read ones), then have a look at [requests](requests.md).
+
+- All transactions are stored in a distributed Ledger (replicated on all Nodes) 
+- The ledger is based on Merkle Tree
+- The ledger consists of two things:
+    - transactions log as a sequence of key-value pairs 
+where key is a sequence number of the transaction and value is the serialized transaction
+    - merkle tree (where hashes for leaves and nodes are persisted)
+- Each transaction has a sequence number (no gaps) - keys in transactions log
+- So, this can be considered as a blockchain where each block size equals to 1
+- There are multiple ledgers by default:
+    - *pool ledger*: transactions related to pool/network configuration (listing all nodes, their keys and addresses)
+    - *config ledger*: transactions for pool configuration plus transactions related to Pool Upgrade
+    - *domain ledger*: all main domain and application specific transactions (including NYM transactions for DID)
+- All transactions are serialized to MsgPack format
+- All transactions (both transaction log and merkle tree hash stores) are stored in LevelDB
+- One can use `read_ledger` script to get transactions for a specified ledger in a readable (JSON) format
+- See [roles and permissions](https://docs.google.com/spreadsheets/d/1TWXF7NtBjSOaUIBeIH77SyZnawfo91cJ_ns4TR-wsq4/edit#gid=0) on the roles and who can create each type of transactions
+
+Below you can find the format and description of all supported transactions.
+
+## Genesis Transactions
+As Indy is Public **Permissioned** blockchain, each ledger may have a number of pre-defined 
+transactions defining the Pool and the Network.
+- pool genesis transactions defining initial trusted Nodes in the Pool
+- domain genesis transactions defining initial trusted Trustees and Stewards
+
+## Common Structure
+Each transaction has the following structure containing of metadata values (common for all transaction types) and 
+transaction specific data:
+```
+{
+    "txn": {
```

Comment:
Its not functionally wrong, but `txn` under transaction seems odd, maybe call it `payload`

Reply:
+1 to switching this to payload, but again comes down to semantics, so it's more of a preference than a necessity.

Reply:
Yes, `payload` is probably better.